waspc/src/Wasp/Analyzer/Parser/Valid.hs

```diff
@@ -0,0 +1,27 @@
+module Wasp.Analyzer.Parser.Valid
+  ( validateAst,
+  )
+where
+
+import Data.List (find)
+import qualified Wasp.Analyzer.Parser as P
+
+validateAst :: P.AST -> Either (String, P.Ctx) P.AST
+validateAst = validateNoEntityDeclInWaspFile
+
+validateNoEntityDeclInWaspFile :: P.AST -> Either (String, P.Ctx) P.AST
+validateNoEntityDeclInWaspFile ast@(P.AST stmts) = case findEntityDecl stmts of
+  Just (P.WithCtx ctx _) -> Left (entitiesNoLongerSupportedError, ctx)
+  Nothing -> Right ast
+  where
+    findEntityDecl :: [P.WithCtx P.Stmt] -> Maybe (P.WithCtx P.Stmt)
+    findEntityDecl = find isEntityDecl
```

Comment:
You can pattern-match the maybe here and return a boolean, no need to propagate it.

You can call the function `includesEntities :: [P.WithCtx P.Stmt] -> Bool`

Reply:
I've adjusted the code a bit 🤷